### ISSUE TITLE
ARM Cross Compile: crypt() not found

### ISSUE THREAD

**Issue by FloatingMind12** [shadow-maint/shadow#717]
I'm trying to cross compile shadow for ARM on an ubuntu host machine.
These are the commands I ran in order:
```Shell
sudo apt-get install gcc-arm-linux-gnueabihf g++-arm-linux-gnueabihf
sudo apt-get -y install ubuntu-dev-tools automake autopoint \
    xsltproc gettext eatmydata expect byacc libtool libbsd-dev \
    pkgconf tmux
git clone https://github.com/shadow-maint/shadow
cd shadow
 ./autogen.sh --host=arm-linux-gnueabi --without-selinux
CC=arm-linux-gnueabihf-gcc CXX=arm-linux-gnueabihf-g++ ./configure --host=armv7l-linux-gnueabihf 
```

But it is always interrupted on the last command with the error
```
checking for crypt in -lcrypt... no
configure: error: crypt() not found
```

I already tried `sudo apt-get install libcrypt-dev` but it still doesn't work

**Comment by jubalh** [comment]
This is a duplicate of https://github.com/shadow-maint/shadow/issues/526 though providing different information.

**Comment by DoomHammer** [comment]
Any suggestions on how to work around this?

**Comment by ikerexxe** [comment]
I'd suggest to use a virtualization environment, either QEMU or QEMU + Docker, to cross-compile shadow in an x86_64 machine for ARM. That should allow to install all ARM dependencies and still have the power of an x86_64 machine to compile.

**Comment by DoomHammer** [comment]
I don't quite get it. How a VM helps with `crypt()` not being found?

**Comment by ikerexxe** [comment]
The VM/Docker approach fixes it because it changes how `apt-get` behaves. On the host machine, `apt-get install libcrypt-dev` fetches a x86_64 library, which the ARM compiler can't use. Inside an emulated ARM container, `apt-get` natively fetches the ARM version of the library and places it in the standard system paths, allowing a normal compilation to succeed instantly without needing to manually manage cross-compiler sysroots.

**Comment by DoomHammer** [comment]
Ah, got it now. My problem is with `shadow-native`, before we start cross-compilation (though the target is ARM).
